exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -569,28 +570,41 @@ func (r *MachinePoolReconciler) waitForMachineCreation(ctx context.Context, mach
 	return nil
 }
 
-func (r *MachinePoolReconciler) getNodeRefMap(ctx context.Context, c client.Client) (map[string]*corev1.Node, error) {
+func (r *MachinePoolReconciler) getNodeRefMap(ctx context.Context, providerIDList []string, c client.Client) (map[string]*corev1.Node, error) {
 	log := ctrl.LoggerFrom(ctx)
 	nodeRefsMap := make(map[string]*corev1.Node)
 	nodeList := corev1.NodeList{}
-	for {
-		if err := c.List(ctx, &nodeList, client.Continue(nodeList.Continue)); err != nil {
+	completeList := true
+	for _, providerID := range providerIDList {
+		if err := c.List(ctx, &nodeList, client.MatchingFields{index.NodeProviderIDField: providerID}); err != nil {
```

Comment:
I'm wondering if it's really more efficient to do the list call for every single Node vs. doing one list call

Reply:
hm yeah i guess i took this for granted since the issue had been accepted and marked as help wanted

any thoughts on how i could go about proving out this efficiency improvement (or lack thereof 😄 )?  or you think it's not worth the effort?

Reply:
Yup sorry I didn't realize this on the issue. It's probably not worth the effort to be honest

Reply:
sure thing, fine with me will close both the PR and the issue then 👍

Reply:
Thx for working on it!!